model/src/main/kotlin/Provenance.kt

```diff
@@ -80,7 +80,9 @@ data class RepositoryProvenance(
     val resolvedRevision: String
 ) : KnownProvenance {
     init {
```

Comment:
Why haven't you decided for keeping the require as-is, and make the tests construct instances with non-empty resolved revision?

I'm asking to me it seems that if repository provenance is sued, the VCS info should not be empty, because otherwise it would be an Unkown provenance, wouldn't it?

Reply:
You are probably correct. Changing the tests¹ to use `"main"` as revision instead of blank, like I did for [model/src/test/kotlin/OrtResultTest.kt](https://github.com/oss-review-toolkit/ort/pull/8764/files?file-filters%5B%5D=.kt&show-viewed-files=true#diff-c52bc83bc22aea9f28992016b2b705f488d2f9f231508bd1705cd1f9b2754c9e), might be the better way to solve the issue with the tests, than allowing blank `resolvedRevision`s on for `VcsInfo.Empty`.

However, the [Analyzer](https://github.com/oss-review-toolkit/ort/pull/8764/commits/2f69343732b3629d16b47feb9edfa3c5cf8f6c80?file-filters%5B%5D=.kt&show-viewed-files=false#diff-35aa45acd6a9c6882501f95739dbb3cb4ec27cf6ec36089d5473b967b8bd2e8f) also allows `VcsInfo.Empty` to be passed from `workingTree` to the `Repository`. And with `VcsInfo.Empty` `workingTree` will very likely also return an empty revision. So in order to not change that behavior of the Analyzer and Repository in this case, I added this exception to `RepositoryProvenance`.

I have not yet noticed any issues regarding non-`Repository` uses of the `RepositoryProvenance`.

Still you are bringing up some good points, maybe the `Analyzer` behavior could change in this case? Then we could revert the exception to `RepositoryProvenance`. We could dial this in once we handle other `Provenance`s in the `Analyzer`.

¹ [evaluator/src/test/kotlin/TestData.kt](https://github.com/oss-review-toolkit/ort/pull/8764/files?file-filters%5B%5D=.kt&show-viewed-files=true#diff-45f6c3132de3d7712b3032b29f22f011ea18979ab3f9f0d2c744354dc4b15a19)
[model/src/test/kotlin/licenses/TestData.kt](https://github.com/oss-review-toolkit/ort/pull/8764/files?file-filters%5B%5D=.kt&show-viewed-files=true#diff-0cd329b641afb2f5622dad4a87a827922b7580fda2e8dad92def8c12c26b3203)
[reporter/src/testFixtures/kotlin/TestData.kt](https://github.com/oss-review-toolkit/ort/pull/8764/files?file-filters%5B%5D=.kt&show-viewed-files=true#diff-00f03b3f88f750595dffb7115e0e27eea71f21c58426e7d1d36b3d54974d9efe)